File: 344. Reverse String/javascript/main.js
/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
var reverseString = function (s) {
    // use for loop
  for (let index = 0; index < s.length / 2; index++) {
    let temp = s[index];
    s[index] = s[s.length - 1 - index];
    s[s.length - 1 - index] = temp;
  }
//   use built-in function
  s.reverse();

  // use two pointers
  let left = 0, right = s.length - 1;
  while (left < right) {
    let temp = s[left];
    s[left] = s[right];
    s[right] = temp;
    left++;
    right--;
  }
};

File: 344. Reverse String/python/main.py
from typing import List

class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        # use built-in function
        s.reverse()
        
        # use two pointers
        # left, right = 0, len(s) - 1
        while left < right:
            s[left], s[right] = s[right], s[left]
            left += 1
            right -= 1
            
        # use for loop
        for i in range(0, len(s)//2):
            s[i], s[-i-1] = s[-i-1], s[i]
            
        return s
            

print(Solution().reverseString(["h","e","l","l","o"]))